src/components/input/Select.tsx

```diff
@@ -112,22 +126,37 @@ function SelectOption<T>({
         },
         classes,
       )}
-      onClick={() => {
-        if (!disabled) {
-          selectOrToggle();
+      onClick={e => {
+        // Do not invoke callback if clicked element is the checkbox, as it has
+        // its own event handler.
+        if (!disabled && e.target !== checkboxRef.current) {
+          selectOneValue();
         }
       }}
-      onKeyPress={e => {
-        if (!disabled && ['Enter', 'Space'].includes(e.code)) {
+      onKeyDown={e => {
+        if (disabled) {
+          return;
+        }
+
+        if (
+          ['Enter', 'Space'].includes(e.code) &&
```

Comment:
Should this use `e.key` instead, ie. the logical rather than physical key that was pressed? I don't know of a keyboard layout where this affects these two keys, but I would guess that it is normally the logical keys that are tested.

Reply:
Right! It should be `key`, not `code`.